Projects/DbCLI/CreateDb.py

```diff
@@ -69,85 +69,94 @@
-parser.add_option(
-  '--nameProp', default='_Name',
-  help='specify the SD property to be used for the molecule names. Default is to use the mol block name')
-parser.add_option(
-  '--missingPropertyVal', default='N/A',
-  help='value to insert in the database if a property value is missing. Default is %default.')
-parser.add_option('--addProps', default=False, action='store_true',
-                  help='add computed properties to the output')
-parser.add_option('--noExtras', default=False, action='store_true',
-                  help='skip all non-molecule databases')
-parser.add_option('--skipLoad', '--skipMols', action="store_false", dest='loadMols', default=True,
-                  help='skip the molecule loading (assumes mol db already exists)')
-parser.add_option('--updateDb', '--update', default=False, action='store_true',
-                  help='add to an existing database')
-parser.add_option('--doPharm2D', default=False, action='store_true',
-                  help='skip calculating Pharm2D fingerprints')
-parser.add_option('--pharm2DTableName', default='pharm2dfps',
-                  help='name of the Pharm2D fingerprints table')
-parser.add_option('--fdefFile', '--fdef',
-                  default=os.path.join(RDConfig.RDDataDir, 'Novartis1.fdef'),
-                  help='provide the name of the fdef file to use for 2d pharmacophores')
-parser.add_option('--doGobbi2D', default=False, action='store_true',
-                  help='skip calculating Gobbi 2D fingerprints')
-parser.add_option('--gobbi2DTableName', default='gobbi2dfps',
-                  help='name of the Gobbi 2D fingerprints table')
-
-parser.add_option('--noMorganFps', '--noCircularFps', default=True, dest='doMorganFps',
-                  action='store_false', help='skip calculating Morgan (circular) fingerprints')
-parser.add_option('--morganFpTableName', default='morganfps',
-                  help='name of the Morgan fingerprints table')
-
-parser.add_option('--delimiter', '--delim', default=' ', help='the delimiter in the input file')
-parser.add_option('--titleLine', default=False, action='store_true',
-                  help='the input file contains a title line')
-parser.add_option('--smilesColumn', '--smilesCol', default=0, type='int',
-                  help='the column index with smiles')
-parser.add_option('--nameColumn', '--nameCol', default=1, type='int',
-                  help='the column index with mol names')
+
+def initParser():
+  """ Initialize the command line parser """ 
+  parser = argparse.ArgumentParser(usage='CreateDB [optional arguments] <filename>',
+                                   description=_description,
+                                   formatter_class=argparse.RawDescriptionHelpFormatter)
+
+  parser.add_argument('filename', nargs='?', help='File containg molecules to load into database')
+  parser.add_argument('--version', action='version', version='%(prog)s ' + _version)
```

Comment:
- positional arguments must be explicitly defined
- `--version` returns version number